### PR TITLE
Iar 19/martin/add authentication

### PR DESCRIPTION
**Pull request by Karhalll**

### CODE REVIEW

**Comment by Karhalll** [comment]
Changes for this ticket are only in log and regist form component. Rest is from merging with Jirinkas IAR-13 that is not yet merged with master.

File: debug.log [hunk 1]
```diff
@@ -15,3 +15,4 @@
 [1130/214333.166:ERROR:directory_reader_win.cc(43)] FindFirstFile: Syst�m nem�e nal�zt uvedenou cestu. (0x3)
```
Comment (SimProch) [inline debug.log:15]: See comment in https://github.com/green-fox-academy/IceAge-Reddit-Frontend/pull/7/files

File: src/app/services/gateway.service.ts [hunk 2]
```diff
@@ -16,10 +18,17 @@ export class GatewayService {
         return this._httpClient.get<PostResponse>(this._data);
     }
 
-    postRegistrationForm(user: User): Observable<User> {
+    postRegistrationForm(user: User): Observable<Token | Error> {
         return this._httpClient.post(
             'http://localhost:3000/api/v1/auth/sign-in',
             user,
-        ) as Observable<User>;
+        ) as Observable<Token>;
```
Comment (SimProch) [inline src/app/services/gateway.service.ts:25]: The typing at the end does not correspond to typing of the function. Why did you omit `Error`?

File: src/types/error.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface Error {
```
Comment (SimProch) [inline src/types/error.ts:1]: You may want to take a look into https://angular.io/api/common/http/HttpErrorResponse. If you have, I'm wondering why was it insufficient to be used?

**Review by SimProch — APPROVED** [review]

**Review by AndyM22 — APPROVED** [review]
All good 👍

**Pull request merged**